### PR TITLE
[WIP][FLINK-10569][tests] Clean up uses of Scheduler and Instance in valid tests(Batch 2)

### PR DESCRIPTION
**Pull request by TisonKun**
## What is the purpose of the change

1. Remove `Instance` usage in ExecutionGraphDeploymentTest

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector:(no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by zentol** [comment]
checkstyle tho:
```
16:29:37.118 [ERROR] src/test/java/org/apache/flink/runtime/executiongraph/utils/SimpleAckingTaskManagerGateway.java:[39,8] (imports) UnusedImports: Unused import: java.util.Optional.
16:29:37.118 [ERROR] src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphTestUtils.java:[28,8] (imports) UnusedImports: Unused import: org.apache.flink.runtime.blob.PermanentBlobService.
16:29:37.118 [ERROR] src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphTestUtils.java:[33,8] (imports) UnusedImports: Unused import: org.apache.flink.runtime.deployment.TaskDeploymentDescriptor.
16:29:37.118 [ERROR] src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphTestUtils.java:[63,8] (imports) UnusedImports: Unused import: akka.actor.Status.
```

**Comment by zentol** [comment]
@flinkbot disapprove quality

File: flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java [hunk 1]
```diff
@@ -740,49 +711,39 @@ public void testFailureWhileRestarting() throws Exception {
 	//  Utilities
 	// ------------------------------------------------------------------------
 
-	private Scheduler createSchedulerWithInstances(int num, TaskManagerGateway taskManagerGateway) {
+	private SlotProvider createSchedulerWithInstances(JobID jobID, int slots) {
```
Comment (TisonKun) [inline flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java]: @tillrohrmann The last usage of Instance in valid tests is in this method. I have no idea how to get rid of it. Could you take a look or give some advice? Thanks!

**Comment by zentol** [comment]
@flinkbot disapprove all

Really, we could've merged this PR as is with the checkstyle fixes. Now we got another 7 commits to review first before anything is merged, that are also not mentioned at all in the description.

**Comment by TisonKun** [comment]
@zentol hmmm, for a better review experience, I would break changes into separated prs.

File: flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java (hunk 1 above)
Comment (TisonKun) [reply flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java]: FYI, I break this pr into separated pr. But this help request is still valid :-)

**Pull request closed without merging**